File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        vector<string> res;
        int n = strs.size();
        int end=0;
        for(int i=0; i<n; i+=end)
        {
            string s = strs[i];
            reverse(s.begin(), s.end());
            
            res.push_back(strs[i]);
            for(int j=i+1; j<n-1; j++)
            {
                if(s==strs[j])
                {
                    res.push_back(strs[j]);
                    end = j;
                }
            }
             ans.push_back(res);   
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hmap = {}
        for i in strs:
            if tuple(sorted(i)) in hmap:
                hmap[tuple(sorted(i))].append(i)
            else:
                hmap[tuple(sorted(i))] = [i]

        res = hmap.values()
        return list(res)
        